### PR TITLE
Update DateToHoliday and DistanceToHoliday primitives to work with timezone-aware inputs

### PR DESCRIPTION
**Pull request by thehomebrewnerd**
This PR updates the `DateToHoliday` and `DistanceToHoliday` primitives to work with timezone-aware inputs and adds new tests to verify the correct answer is returned.

### CODE REVIEW

File: featuretools/primitives/standard/datetime_transform_primitives.py [hunk 1]
```diff
@@ -201,7 +201,7 @@ def distance_to_holiday(x):
             df["x_index"] = df.index  # store original index as a column
             df = df.dropna()
             df = df.sort_values("date")
-            df.date = df.date.dt.normalize()
+            df.date = df.date.dt.normalize().dt.date.astype("datetime64[ns]")
```
Comment (rwedge) [inline featuretools/primitives/standard/datetime_transform_primitives.py:204]: do we need to normalize if we are also extracting the date?

File: featuretools/tests/primitive_tests/test_datetoholiday_primitive.py [hunk 2]
```diff
@@ -109,3 +109,26 @@ def test_multiple_countries():
...
+
+def test_with_timezone_aware_datetimes():
+    df = pd.DataFrame(
+        {
+            "non_timezone_aware_with_time": pd.date_range(
+                "2018-07-03 09:00", periods=3
+            ),
+            "non_timezone_aware_no_time": pd.date_range("2018-07-03", periods=3),
+            "timezone_aware_with_time": pd.date_range(
+                "2018-07-03 09:00", periods=3
```
Comment (rwedge) [inline featuretools/tests/primitive_tests/test_datetoholiday_primitive.py:122]: should we test with times where if converted to UTC, some times would change dates?

File: featuretools/primitives/standard/datetime_transform_primitives.py (hunk 1 above)
Comment (thehomebrewnerd) [reply featuretools/primitives/standard/datetime_transform_primitives.py]: No, the normalize calls aren't necessary anymore. Removed. Also updated to (I think) improve readability of these lines a little bit.

**Review by rwedge — APPROVED** [review]
Looks good

**Pull request merged**